rotkehlchen/db/filtering.py

```diff
@@ -1537,3 +1545,132 @@ def make(
+            limit: Optional[int] = None,
+            offset: Optional[int] = None,
+            optional_chain_addresses: Optional[list[OptionalChainAddress]] = None,
+            names: Optional[list[str]] = None,
+    ) -> 'BaseAddressbookFilterQuery':
+        ...
+
+    @abstractmethod
+    def get_book_type(self) -> AddressbookType:
+        ...
+
+    @abstractmethod
+    def get_base_query(self) -> str:
+        ...
+
+
+class UserAddressbookFilterQuery(BaseAddressbookFilterQuery):
+    """Filter to find entries in global address book"""
+    @classmethod
+    def make(  # type: ignore[override]
+            cls,
+            limit: Optional[int] = None,
+            offset: Optional[int] = None,
+            order_by_rules: Optional[list[tuple[str, bool]]] = None,
+            optional_chain_addresses: Optional[list[OptionalChainAddress]] = None,
+            names: Optional[list[str]] = None,
+    ) -> 'UserAddressbookFilterQuery':
+        filter_query = cls.create(
+            and_op=True,
+            limit=limit,
+            offset=offset,
+            order_by_rules=order_by_rules,
+        )
+        filter_query = cast('UserAddressbookFilterQuery', filter_query)
+
+        filters: list[DBFilter] = []
+        if optional_chain_addresses is not None:
+            addresses_filters: list[Any] = []
+            for chain_address in optional_chain_addresses:
+                if chain_address.blockchain is None:
+                    addresses_filters.append(DBEqualsFilter(
+                        and_op=True,
+                        column='address',
+                        value=chain_address.address,
+                    ))
+                else:
+                    addresses_filters.append(DBNestedFilter(
+                        and_op=True,
+                        filters=[
+                            DBEqualsFilter(
+                                and_op=True,
+                                column='address',
+                                value=chain_address.address,
+                            ),
+                            DBEqualsFilter(
+                                and_op=True,
+                                column='blockchain',
+                                value=chain_address.blockchain.value,
+                            ),
+                        ],
+                    ))
+            filters.append(DBNestedFilter(
+                and_op=False,
+                filters=addresses_filters,
+            ))
+        if names is not None:
+            filters.append(DBMultiStringFilter(
+                and_op=True,
+                column='name',
+                operator='IN',
+                values=names,
+            ))
+
+        filter_query.filters = filters
+        return filter_query
+
+    def get_book_type(self) -> AddressbookType:
+        return AddressbookType.USER
+
+    def get_base_query(self) -> str:
+        return 'SELECT address, name, blockchain FROM address_book '
```

Comment:
this is not the right place for those since you are mixing concerns. One thing is the filters and other is the structures that is related to. Those can be functions that check the class type and return the proper value.